internal/discovery/saptune.go

```diff
@@ -0,0 +1,80 @@
+package discovery
+
+import (
+	"encoding/json"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/internal/core/saptune"
+	"github.com/trento-project/agent/internal/discovery/collector"
+	"github.com/trento-project/agent/pkg/utils"
+)
+
+const SaptuneDiscoveryID string = "saptune_discovery"
+const SaptuneDiscoveryMinPeriod time.Duration = 1 * time.Second
+
+type SaptuneDiscovery struct {
+	id              string
+	collectorClient collector.Client
+	interval        time.Duration
+}
+
+type SaptuneDiscoveryPayload struct {
+	PackageVersion   string      `json:"package_version"`
+	SaptuneInstalled bool        `json:"saptune_installed"`
+	Status           interface{} `json:"status"`
+}
+
+func NewSaptuneDiscovery(collectorClient collector.Client, config DiscoveriesConfig) Discovery {
+	return SaptuneDiscovery{
+		id:              SaptuneDiscoveryID,
+		collectorClient: collectorClient,
+		interval:        config.DiscoveriesPeriodsConfig.Saptune,
+	}
+}
+
+func (d SaptuneDiscovery) GetID() string {
+	return d.id
+}
+
+func (d SaptuneDiscovery) GetInterval() time.Duration {
+	return d.interval
+}
+
+func (d SaptuneDiscovery) Discover() (string, error) {
+	var saptunePayload SaptuneDiscoveryPayload
+
+	saptuneRetriever, err := saptune.NewSaptune(utils.Executor{})
+	switch {
+	case err != nil:
+		saptunePayload = SaptuneDiscoveryPayload{
+			PackageVersion:   "",
+			SaptuneInstalled: false,
+			Status:           nil,
+		}
+	case !saptuneRetriever.IsJSONSupported:
+		saptunePayload = SaptuneDiscoveryPayload{
+			PackageVersion:   saptuneRetriever.Version,
+			SaptuneInstalled: true,
+			Status:           nil,
+		}
+	default:
+		saptuneData, _ := saptuneRetriever.RunCommandJSON("status")
+		unmarshalled := make(map[string]interface{})
+		_ = json.Unmarshal(saptuneData, &unmarshalled)
```

Comment:
I think we should capture the error and exit if there is any.
Just for safetiness. 
The status should always return  a json, but just in case...

Reply:
The thing is that as you said, some `saptune` commands will return non-exit codes which still have valid data. If we exit we wouldn't return that JSON, right? or do you mean to simply:

```
		if err != nil {
			log.Debugf("Error while unmarshalling saptune status: %s", err)
		}
```
and thats it?